src/main.go

```
@@ -1,126 +1,182 @@
+	/*A defer statement defers the execution of a function until the surrounding function returns.
+	simply, run cur.Close() process but after cur.Next() finished.*/
+	defer cur.Close(context.TODO())
+
+	for cur.Next(context.TODO()) {
+
+		// create a value into which the single document can be decoded
+		var book models.Book
+		// & character returns the memory address of the following variable.
+		err := cur.Decode(&book) // decode similar to deserialize process.
+		if err != nil {
+			log.Fatal(err)
 		}
+
+		// add item our array
+		books = append(books, book)
+	}
+
+	if err := cur.Err(); err != nil {
+		log.Fatal(err)
 	}
+
+	json.NewEncoder(w).Encode(books) // encode similar to serialize process.
 }
 
-func createTask(w http.ResponseWriter, r *http.Request)  {
-	var newTask task
-	reqBody,err := ioutil.ReadAll(r.Body)
+func getBook(w http.ResponseWriter, r *http.Request) {
+
+	w.Header().Set("Content-Type", "application/json")
+
+	var book models.Book
+	// we get params with mux.
+	var params = mux.Vars(r)
+
+	// string to primitive.ObjectID
+	id, _ := primitive.ObjectIDFromHex(params["id"])
+
+	collection := helper.ConnectDB()
+
+	// We create filter. If it is unnecessary to sort data for you, you can use bson.M{}
+	filter := bson.M{"_id": id}
+	err := collection.FindOne(context.TODO(), filter).Decode(&book)
 
 	if err != nil {
-		fmt.Fprintf(w, "Not valid objetc")
+		helper.GetError(err, w)
+		return
 	}
 
-	json.Unmarshal(reqBody, &newTask)
-	newTask.Id = len(tasks) + 1
-	tasks = append(tasks, newTask)
+	json.NewEncoder(w).Encode(book)
+}
 
+func createBook(w http.ResponseWriter, r *http.Request) {
 	w.Header().Set("Content-Type", "application/json")
-	w.WriteHeader(http.StatusCreated)
-	json.NewEncoder(w).Encode(newTask)
-}
 
-func deleteTask(w http.ResponseWriter, r *http.Request)  {
-	vars := mux.Vars(r)
-	taskId, err := strconv.Atoi(vars["id"])
+	var book models.Book
+
+	// we decode our body request params
+	_ = json.NewDecoder(r.Body).Decode(&book)
```

Never ignore errors.